nilearn/_utils/cache.py

```diff
@@ -132,7 +131,7 @@ def cache(func, memory, ref_memory_level=2, memory_level=1, **kwargs):
         returned.
     """
 
-    if ref_memory_level <= memory_level or memory is None:
+    if ref_memory_level > memory_level or memory is None:
```

Comment:
This is weird. Are these parameters used somewhere? My impression is that the logic is in the mixin and that this function is always called with the default parameters.